test/test_dynamic_shapes.py

```diff
@@ -222,6 +222,162 @@ def test_unsqueeze_copy_dynamism(self):
+    self.assertRaises(RuntimeError, lambda: t3.view(t2.shape[0]))
+
+    # If their “dynamic” values are incompatible, a RuntimeError is raised.
+    t4 = torch.randint(10, (2, 3), device=dev)
+    self.assertRaises(RuntimeError, lambda: t4.view(t2.shape[0]))
+
+    # verify if dynamism is propagated correctly.
+    t5 = torch.tensor([1, 1, 3, 5, 1, 6], device=dev)
+    t6 = torch.nonzero(t5)
+    t7 = torch.randint(10, (2, 3), device=dev)
+    t8 = t7.view(t6.shape[0])
+    self.assertIsInstance(t8.shape[0], torch.SymInt)
+    self.assertEqual(str(t8.shape[0]), '<=6')
+    self.assertEqual(t8.shape[0], 6)
+
+    # verify correctness.
+    t5_aten = torch.tensor([1, 1, 3, 5, 1, 6])
+    t6_aten = torch.nonzero(t5_aten)
+    t7_aten = torch.randint(10, (2, 3))
+    t8_aten = t7_aten.view(t6_aten.shape[0])
+    self.assertEqual(t8.cpu(), t8_aten.cpu())
+
+  def test_view_copy_symint_with_dyn_input_static_input_shape(self):
+    # If the input tensor and shape are “statically” incompatible, a compilation error is raised.
+    t1 = torch.tensor([1, 0, 3, 5, 0, 6], device=dev)
+    # t2.shape=torch.Size([<=6, 1]) with real size [4, 1]
+    # t2 = [[0], [2], [3], [5]]
+    t2 = torch.nonzero(t1)
+    # static 6 is incompatible with 7
+    self.assertRaises(RuntimeError, lambda: t2.view(7))
+
+    # If their “dynamic” values are incompatible, a RuntimeError is raised.
+    t4 = torch.randint(10, (2, 3), device=dev)
+    # statically compatible.
+    # It will fail in pytorch layer.
+    self.assertRaises(RuntimeError, lambda: t2.view(6))
+
+    # verify if dynamism is propagated correctly.
+    t5 = torch.tensor([1, 1, 3, 5, 1, 6], device=dev)
+    t6 = torch.nonzero(t5)
+    t7 = t6.view(6)
+    self.assertIsInstance(t7.shape[0], torch.SymInt)
+    self.assertEqual(str(t7.shape[0]), '<=6')
+    self.assertEqual(t7.shape[0], 6)
+
+    # verify correctness.
+    t5_aten = torch.tensor([1, 1, 3, 5, 1, 6])
+    t6_aten = torch.nonzero(t5_aten)
+    t7_aten = t6_aten.view(6)
+    self.assertEqual(t7.cpu(), t7_aten.cpu())
+
+  def test_view_copy_symint_with_dyn_input_dyn_input_shape(self):
+    # If the input tensor and shape are “statically” incompatible, a compilation error is raised.
+    t1 = torch.tensor([1, 0, 3, 5, 0, 6], device=dev)
+    # t2.shape=torch.Size([<=6, 1]) with real size [4, 1]
+    # t2 = [[0], [2], [3], [5]]
+    t2 = torch.nonzero(t1)
+    t3 = torch.tensor([1, 0, 3, 5, 0, 6, 7], device=dev)
+    # t4.shape=torch.Size([<=7, 1]) with real size [5, 1]
+    t4 = torch.nonzero(t3)
+    self.assertRaises(RuntimeError, lambda: t2.view(t4.shape[0]))
+
+    # If their “dynamic” values are incompatible, a RuntimeError is raised.
+    t5 = torch.tensor([1, 2, 3, 4, 5, 6, 0], device=dev)
+    # t6.shape=torch.Size([<=7, 1]) with real size [6, 1]
+    t6 = torch.nonzero(t5)
+    # statically compatible but dynamically incompatible.
+    # It will fail in pytorch layer.
+    self.assertRaises(RuntimeError, lambda: t6.view(t4.shape[0]))
```

Comment:
comment to me: it fails in pytorch layer: https://paste.googleplex.com/5104119952965632